### ISSUE TITLE
Heatmap with phyloseq object

### ISSUE THREAD

**Issue by sewuneta** [joey711/phyloseq#1422]
Dear all,
I was following the phyloseq tutorial to create a heatmap. But with three soil and 12 plant genotypes as factor the figure was not too explanatory. Can someone here help to create a heat map like this? (with my two factors layered on top of the heatmap?) 
![image](https://user-images.githubusercontent.com/21764012/102207992-e83ef000-3edf-11eb-9fd6-720a9ae16791.png)

**Comment by jyotsnepal** [comment]
Hi Did you manage to get it? Could you please help if you know already

**Comment by jyotsnepal** [comment]
Yes please, that would be a great help, I also have 11 genotypes and 3 soils as factors.

**Comment by Sewunet-Abera** [comment]
Here you go.
I'm a bit working on something urgent; otherwise I should have prepared a nice rmarkdown file with clear comments.
But still you can go through this and if you bump into a jurgen code let me know.

Greetings


#plot the heat map
##First create the numeric data matrix by subsetting the genes name
####upset.core.top100.50.hmap is my dataframe with col.names as samples and rownames as genes

df_num_fun <- as.matrix(upset.core.top100.50.hmap[,2:13])
head(df_num_fun)
rownames(df_num_fun)

#Then renomve NA
df_num_fun <- na.omit(df_num)


#Avoiding numeric index from the row names
rownames(df_num_fun) = sapply(upset.core.top100.50.hmap$Name,function(x)
  strsplit(as.character(x),split = "\\\\")[[1]][1])


#scale the data matrix
df_num_fun_scale = scale(df_num_fun)
head(df_num_fun_scale)
dim(df_num_fun_scale)

#the default pheatmap.
pheatmap(df_num_fun_scale,
         main = "pheatmap default")


#with the row scaling
pheatmap(df_num_fun_scale,
         scale = "row",
         main = "pheatmap row scaling")


#Annotate the columns and rows
path_df <- data.frame("Genes" = upset.core.top100.50.hmap$Name)
head(path_df)


rownames(path_df) = rownames(df_num_fun_scale) # name matching

pheatmap(df_num_scale,
         cluster_cols = T,
         #annotation_row = path_df,
         main = "pheatmap row annotation")


#column scaling
#make a data frame for the heatmap
meta.data.e40 <- read.csv("meta.data.e40.csv",
                          header = T,
                          sep=",")
head(meta.data.e40)
dim(meta.data.e40)
ph.data <- meta.data.e40[,2:8]

row.names(ph.data) <- NULL
df <- as.data.frame(ph.data)

head(df)


#Column annotation
my_sample_col <- df %>%
  select(-c(Sample_Type,
            Genotype, St_Rx,
            LGS, DarT_Cluster,
            Domestication)) %>%
  filter(Phenotype!="Bulk_Soil")

my_sample_col <- df %>%
  select(Phenotype) %>%
  filter(Phenotype!="Bulk_Soil")

head(my_sample_col)


# name matching
row.names(my_sample_col) <- colnames(df_num)

pheatmap(df_num_fun_scale,
         clustering_method = "complete",
         show_rownames=T,
         cluster_rows = T,
         #annotation_col = my_sample_col,
         cluster_cols = T,
         #annotation_row = path_df,
         cutree_rows = 10,
         cutree_cols = 8,
         fontsize = 7,
         display_numbers = T,
         number_color = "black",
         fontsize_number = 5,
         legend = T)


ComplexHeatmap::pheatmap(df_num_fun_scale)


p <- pheatmap(df_num_fun_scale,
              clustering_method = "complete",
              show_rownames=T,
              cluster_rows = T,
              annotation_col = my_sample_col,
              cluster_cols = T,
              #annotation_row = path_df,
              cutree_rows = 10,
              cutree_cols = 8,
              fontsize = 7,
              display_numbers = T,
              number_color = "black",
              fontsize_number = 5,
              #cellwidth = 16, cellheight = 12)
              legend = F,
              clustering_distance_rows = "correlation")


draw(p, legend_grouping = "original")


________________________________
